x-pack/legacy/plugins/searchprofiler/public/np_ready/application/components/profile_tree/shard_details/shard_details_tree_node.tsx

```diff
@@ -0,0 +1,119 @@
+  const [childrenVisible, setChildrenVisible] = useState(hasVisibleChild(operation));
+  const { highlight, isHighlighted, id } = useHighlightTreeNode();
+
+  const renderTimeRow = (op: Operation) => (
+    <div className="prfDevTool__profileTree__tvRow">
+      <div className="prfDevTool__profileTree__cell prfDevTool__profileTree__description">
+        <EuiLink
+          className="prfDevTool__profileTree__shardDetails"
+          disabled={!op.hasChildren}
+          onClick={() => setChildrenVisible(!childrenVisible)}
+        >
+          {op.hasChildren ? (
+            <EuiIcon type={childrenVisible ? 'arrowDown' : 'arrowRight'} />
+          ) : (
+            // Use dot icon for alignment if arrow isn't there
+            <EuiIcon type={'dot'} />
+          )}
+
+          {op.query_type}
+        </EuiLink>
+      </div>
+      {/* Self Time Badge */}
+      <div className="prfDevTool__profileTree__cell prfDevTool__profileTree__time">
+        <EuiBadge
+          className="prfDevTool__profileTree__badge"
+          style={{ backgroundColor: op.absoluteColor }}
+        >
+          {msToPretty(op.selfTime!, 1)}
+        </EuiBadge>
+      </div>
+      {/* Total Time Badge */}
+      <div className="prfDevTool__profileTree__cell prfDevTool__profileTree__totalTime">
+        <EuiBadge
+          className="prfDevTool__profileTree__badge"
+          style={{ backgroundColor: op.absoluteColor }}
+        >
+          {msToPretty(op.time!, 1)}
+        </EuiBadge>
+      </div>
+      {/* Time percentage Badge */}
+      <div className="prfDevTool__profileTree__cell prfDevTool__profileTree__percentage">
+        <EuiBadge
+          className="prfDevTool__profileTree__progress--percent"
+          style={{ '--prfDevToolProgressPercentage': op.timePercentage } as any}
+        >
+          <span
+            className="prfDevTool__progress--percent-ie"
+            style={{ width: op.timePercentage + '%' }}
+          />
+          <span className="prfDevTool__progressText">{op.timePercentage + '%'}</span>
+        </EuiBadge>
+      </div>
+    </div>
+  );
+
+  return (
+    <>
+      <div
+        key={id}
+        className={isHighlighted() ? 'prfDevTool__tvRow--last' : ''}
+        style={{ paddingLeft: operation.depth! * TAB_WIDTH_PX + 'px' }}
+      >
+        <div className="prfDevTool__profileTree__tvRow">{renderTimeRow(operation)}</div>
+        <div className="prfDevTool__profileTree__tvRow">
+          <span className="prfDevTool__detail">
+            <EuiCodeBlock paddingSize="none">
+              {limitString(operation.lucene || '', 120)}
+            </EuiCodeBlock>
+            <EuiLink
+              type="button"
+              onClick={() => highlight({ indexName: index.name, operation, shard })}
+            >
+              {i18n.translate('xpack.searchProfiler.profileTree.body.viewDetailsLabel', {
+                defaultMessage: 'View Details',
```

Comment:
nit: I think this should be sentence case

https://elastic.github.io/eui/#/guidelines/writing